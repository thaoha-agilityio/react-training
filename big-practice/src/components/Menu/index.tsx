import { memo } from 'react';
import { Link } from 'react-router-dom';
import isEqual from 'react-fast-compare';

// Components
import { ListItem, Text, UnorderedList } from '@chakra-ui/react';

// Types
import { IMenuItem } from '@types';

type MenuProps = {
  menu: IMenuItem[];
};

const Menu = ({ menu }: MenuProps): JSX.Element => (
  <UnorderedList display='flex' listStyleType='none' gap={{ base: '10px', md: '25px' }}>
    {menu.map((item) => (
      <ListItem key={item.title}>
        <Link to={item.path}>
          <Text fontSize={{ base: 'xs', md: 'sm' }} color='gray.600'>
            {item.title}
          </Text>
        </Link>
      </ListItem>
    ))}
  </UnorderedList>
);

export default memo(Menu, isEqual);
