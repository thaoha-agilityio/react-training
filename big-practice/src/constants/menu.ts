import { ROUTES } from './routers';

export const MENU = [
  { title: 'Home', path: ROUTES.HOMEPAGE },
  { title: 'Shop', path: ROUTES.SHOP },
  { title: 'Add Product', path: ROUTES.ADD_PRODUCT },
];

export const HELPS = [
  { title: 'Payment Options', path: '/' },
  { title: 'Returns', path: '/' },
  { title: 'Privacy Policies', path: '/' },
];
