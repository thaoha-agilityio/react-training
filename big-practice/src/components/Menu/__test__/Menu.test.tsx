import Menu from '../index';

// Constants
import { MENU } from '@constants/menu';

// Helpers
import { renderWithRouterAndQuery } from '@helpers';

describe('Menu component', () => {
  it('renders a list of menu items with links', () => {
    const { container, getByRole } = renderWithRouterAndQuery(<Menu menu={MENU} />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();

    // Check that each menu item and link is rendered
    MENU.forEach((item) => {
      const linkElement = getByRole('link', { name: item.title });
      expect(linkElement).toBeInTheDocument();
      expect(linkElement).toHaveAttribute('href', item.path);
    });
  });
});
