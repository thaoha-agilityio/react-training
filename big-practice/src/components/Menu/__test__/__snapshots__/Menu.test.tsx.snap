// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders a list of menu items with links 1`] = `
<div>
  <ul
    class="css-1a82tfq"
    role="list"
  >
    <li
      class="css-0"
    >
      <a
        href="/"
      >
        <p
          class="chakra-text css-2zbmko"
        >
          Home
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/shop"
      >
        <p
          class="chakra-text css-2zbmko"
        >
          Shop
        </p>
      </a>
    </li>
    <li
      class="css-0"
    >
      <a
        href="/add-product"
      >
        <p
          class="chakra-text css-2zbmko"
        >
          Add Product
        </p>
      </a>
    </li>
  </ul>
</div>
`;
